driver-kotlin-coroutine/src/main/kotlin/com/mongodb/kotlin/client/coroutine/MongoClient.kt

```diff
@@ -0,0 +1,287 @@
+        ): MongoClient {
+            return create(
+                MongoClientSettings.builder().applyConnectionString(connectionString).build(), mongoDriverInformation)
+        }
+
+        /**
+         * Create a new client with the given connection string.
+         *
+         * For each of the settings classed configurable via [MongoClientSettings], the connection string is applied by
+         * calling the `applyConnectionString` method on an instance of setting's builder class, building the setting,
+         * and adding it to an instance of [com.mongodb.MongoClientSettings.Builder].
+         *
+         * @param settings the client settings
+         * @param mongoDriverInformation any driver information to associate with the MongoClient
+         * @return
+         */
+        public fun create(
+            settings: MongoClientSettings,
+            @Nullable mongoDriverInformation: MongoDriverInformation? = null
+        ): MongoClient {
+            val builder =
+                if (mongoDriverInformation == null) MongoDriverInformation.builder()
+                else MongoDriverInformation.builder(mongoDriverInformation)
+            return MongoClient(JMongoClients.create(settings, builder.driverName("kotlin").build()))
+        }
+    }
+
+    public override fun close(): Unit = wrapped.close()
+
+    /**
+     * Gets the current cluster description.
+     *
+     * This method will not block, meaning that it may return a [ClusterDescription] whose `clusterType` is unknown and
+     * whose [com.mongodb.connection.ServerDescription]s are all in the connecting state. If the application requires
+     * notifications after the driver has connected to a member of the cluster, it should register a
+     * [com.mongodb.event.ClusterListener] via the [com.mongodb.connection.ClusterSettings] in
+     * [com.mongodb.MongoClientSettings].
+     *
+     * @return the current cluster description
+     * @see com.mongodb.connection.ClusterSettings.Builder.addClusterListener
+     * @see com.mongodb.MongoClientSettings.Builder.applyToClusterSettings
+     */
+    public fun getClusterDescription(): ClusterDescription = wrapped.clusterDescription
+
+    /**
+     * Gets a [MongoDatabase] instance for the given database name.
+     *
+     * @param databaseName the name of the database to retrievecom.mongodb.connection.
+     * @return a `MongoDatabase` representing the specified database
+     * @throws IllegalArgumentException if databaseName is invalid
+     * @see com.mongodb.MongoNamespace.checkDatabaseNameValidity
+     */
+    public fun getDatabase(databaseName: String): MongoDatabase = MongoDatabase(wrapped.getDatabase(databaseName))
+
+    /**
+     * Creates a client session.
+     *
+     * Note: A ClientSession instance can not be used concurrently in multiple operations.
+     *
+     * @param options the options for the client session
+     * @return the client session
+     */
+    public suspend fun startSession(
```

Comment:
It would be inconsistent, but if you want you can make this non-suspendable since startSession no longer does I/O.  But maybe better to just keep it consistent.

Reply:
Opting to leave as is because we'd still need to handle / convert the underlying publisher.